File: TransitionPractice/app/src/main/java/com/gandan/android/transitionpractice/MainActivity.java
package com.gandan.android.transitionpractice;

import android.app.SharedElementCallback;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerMain = findViewById(R.id.recyclerMain);
        recyclerMain.setLayoutManager(new LinearLayoutManager(this));
        recyclerMain.setAdapter(new RecycleAdapter(this));


    }
}

File: DiffUtil/app/src/main/java/com/gandan/android/diffutil/MainActivity.kt
package com.gandan.android.diffutil

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.selection.SelectionPredicates
import androidx.recyclerview.selection.SelectionTracker
import androidx.recyclerview.selection.StableIdKeyProvider
import androidx.recyclerview.selection.StorageStrategy
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var list : ArrayList<DataModels.DummyModel> = arrayListOf()
        var recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        var recyclerAdapter = RecyclerAdapter(list)
        recyclerView.apply {
            layoutManager = LinearLayoutManager(context)
            adapter = recyclerAdapter
        }


        for(i in 0..300){
            var model : DataModels.DummyModel = DataModels.DummyModel(i, "아이템 $i")
            list.add(model)
        }
        recyclerView.adapter!!.notifyDataSetChanged()

        var selectionTracker : SelectionTracker<Long> = SelectionTracker.Builder(
            "selection-demo",
            recyclerView,
            StableIdKeyProvider(recyclerView),
            MyItmDetailsLookup(recyclerView),
            StorageStrategy.createLongStorage()).withSelectionPredicate(
            SelectionPredicates.createSelectSingleAnything()
        ).build()


        recyclerAdapter.tracker = selectionTracker
    }
}
